OSVR-Unity/Assets/OSVRUnity/src/K1RadialDistortion.cs

```diff
@@ -0,0 +1,147 @@
+/// OSVR-Unity Connection
+///
+/// http://sensics.com/osvr
+///
+/// <copyright>
+/// Copyright 2014 Sensics, Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+/// </copyright>
+/// <summary>
+/// Long ago, based on Unity's (Pro Only) Image Effect: Fisheye.js
+/// Author: Greg Aring
+/// Email: greg@sensics.com
+/// </summary>
+using UnityEngine;
+
+namespace OSVR.Unity
+{
+	[ExecuteInEditMode]
+	[RequireComponent(typeof(Camera))]
+	public class K1RadialDistortion : MonoBehaviour
+	{
+		public float k1Red = 0.0f;
+		public float k1Green = 0.0f;
+		public float k1Blue = 0.0f;
+		public Vector2 center = new Vector2(0.5f, 0.5f);
+		public Material DistortionMaterial;
+
+		private void OnRenderImage(RenderTexture source, RenderTexture destination)
+		{
+			DistortionMaterial.SetFloat("_K1_Red", k1Red);
+			DistortionMaterial.SetFloat("_K1_Green", k1Green);
+			DistortionMaterial.SetFloat("_K1_Blue", k1Blue);
+			DistortionMaterial.SetVector("_Center", center);
+			Graphics.Blit(source, destination, DistortionMaterial);
+		}
+	}
+
+	public class K1RadialDistortionFactory
+	{
+		private const string ShaderName = "Osvr/OsvrDistortion";
+
+		public bool Supported
+		{
+			get
+			{
+				return _supported;
+			}
+			private set
+			{
+				_supported = value;
+			}
+		}
+
+		private bool _supported = false;
+
+		private Shader DistortionShader;
+
+		public K1RadialDistortionFactory()
+		{
+			Supported = DoSetup();
+			if (!Supported)
+			{
+				Debug.Log("Execution will proceed, but without shader-based distortion.");
+			}
+		}
+
+		/// <summary>
+		/// Creates a K1RadialDistortion effect and adds it as a component to the given eye, if possible.
+		/// </summary>
+		/// <param name="Eye">VREye to apply the effect to.</param>
+		/// <returns>K1RadialDistortion object for parameter setting, or null if not supported</returns>
+		public K1RadialDistortion GetOrCreateDistortion(OSVR.Unity.VREye eye)
+		{
+			K1RadialDistortion ret = eye.DistortionEffect;
+			if (!Supported)
+			{
+				if (ret) {
```

Comment:
If that's the C# idiom, certainly fine with me. Heck, even in C++ I go back and forth between `if (ptr)` and `if (nullptr != ptr)` - the former being typically nicer because it works with all smart pointers, even those without comparison to `nullptr` handled.